src/test/java/com/puppycrawl/tools/checkstyle/gui/MainFrameModelTest.java

```diff
@@ -0,0 +1,156 @@
+
+        // Open valid file with parseMode:PLAIN_JAVA
+
+        model.openFile(javaFile);
+
+        verifyCorrectTestDataInFrameModel(model, javaFile);
+
+        // Open valid file with parseMode:JAVA_WITH_COMMENTS
+
+        model.setParseMode(ParseMode.JAVA_WITH_COMMENTS);
+        model.openFile(javaFile);
+
+        verifyCorrectTestDataInFrameModel(model, javaFile);
+
+        // Open valid file with parseMode:JAVA_WITH_JAVADOC_AND_COMMENTS
+
+        model.setParseMode(ParseMode.JAVA_WITH_JAVADOC_AND_COMMENTS);
+        model.openFile(javaFile);
+
+        verifyCorrectTestDataInFrameModel(model, javaFile);
+
+        // Null parameter
+
+        model.openFile(null);
+
+        verifyCorrectTestDataInFrameModel(model, javaFile);
+
+        // Open noneExisit file
+
+        final File nonExistentFile = new File(getPath(FILE_NAME_NON_EXISTENT));
+
+        try {
+            model.openFile(nonExistentFile);
+        } catch (CheckstyleException ex) {
+            final String expectedMsg = String.format(Locale.ROOT,
+                    "FileNotFoundException occurred while opening file %s.",
+                    nonExistentFile.getPath());
+            assertEquals(expectedMsg, ex.getMessage());
+        }
+
+        assertEquals(nonExistentFile, model.getCurrentFile());
+        assertEquals(getGuiTitleForTestFile(FILE_NAME_NON_EXISTENT), model.getTitle());
+
+        // Open non-compilable file
+
+        final String noneCompilableFilePath = getNonCompilableFilePath();
+        final File nonCompilable = new File(noneCompilableFilePath);
+
+        try {
+            model.openFile(nonCompilable);
+        } catch (CheckstyleException ex) {
+            final String expectedMsg = String.format(Locale.ROOT,
+                    "NoViableAltException occurred while opening file %s.",
+                    nonCompilable.getPath());
+            assertEquals(expectedMsg, ex.getMessage());
+        }
+
+        assertEquals(nonCompilable, model.getCurrentFile());
+        assertEquals(getGuiTitleForTestFile(FILE_NAME_NON_COMPIABLE), model.getTitle());
+    }
+
+    private void verifyCorrectTestDataInFrameModel(MainFrameModel model, File javaFile) {
+        assertEquals(javaFile, model.getCurrentFile());
+        assertEquals(getGuiTitleForTestFile(FILE_NAME_TEST_DATA), model.getTitle());
+        assertTrue(model.isReloadActionEnabled());
+        assertEquals(17, model.getLinesToPosition().size());
+        assertTrue(model.getText().contains("InputJavadocAttributesAndMethods"));
+        assertEquals(getPath(""), model.getLastDirectory() + File.separator);
+        assertNotNull(model.getParseTreeTableModel());
+    }
+
+    private static String getGuiTitleForTestFile(String fileName) {
+        return "Checkstyle GUI : " + fileName;
+    }
+
+    private static String getNonCompilableFilePath() {
+        return "src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/"
```

Comment:
Test cases for `filefilters` and `MainTest.java` are use `InputIncorrectClass.java`.  In `noncompilable` folder, it doesn't have a separate folder for `filters`.